src/firecracker/examples/uffd/uffd_utils.rs

```diff
@@ -55,7 +55,7 @@ pub enum MemPageState {
 }
 
 impl UffdPfHandler {
-    pub fn from_unix_stream(stream: UnixStream, backing_buffer: *const u8, size: usize) -> Self {
+    pub fn from_unix_stream(stream: &UnixStream, backing_buffer: *const u8, size: usize) -> Self {
```

Comment:
```suggestion
    pub fn from_unix_stream<S: AsRef<UnixStream>>(stream: S, backing_buffer: *const u8, size: usize) -> Self {
```
this means the later change
```diff
-     UffdPfHandler::from_unix_stream(stream, memfile_buffer, size)
+    UffdPfHandler::from_unix_stream(&stream, memfile_buffer, size)
```
is not required.

Reply:
I would prefer to stick with `&UnixStream`. There is not point of adding additional abstractions.